docs/nodemcu-lrm.md

```diff
@@ -0,0 +1,335 @@
+
+`lua_debugbreak()` and `ASSERT(condtion)` are available for development debugging.  If `DEVELOPMENT_USE_GDB` is defined then these will respectively trigger a debugger break and evaluate a conditional assert prologue on the same.  If not, then these are effectively ignored and generate no executable code.
+
+#### lua_dump
+
+` int lua_dump (lua_State *L, lua_Writer writer, void *data, int strip);`         [-0, +0, –]
+
+Dumps function at the top of the stack function as a binary chunk as per LRM.  However the last argument is now an integer is in the range -1..2, rather a boolean as per standard Lua:
+- `strip`
+    - -1, use the current default strip level (which can be set by [`lua_striptdebug`](#lua_striptdebug))
+    - 0, keep all debug info
+    - 1, discard Local and Upvalue debug info; keep line number info
+    - 2, discard Local, Upvalue and line number debug info
+
+The internal NodeMCU `Proto` encoding of debug line number information is typically 15× more compact than in standard Lua; the intermediate `strip=1` argument allows the removal of most of the debug information whilst retaining the ability to produce a proper line number traceback on error. 
+
+#### lua_freeheap
+
+` int lua_freeheap (void);`         [-0, +0, –]
+
+returns the amount of free heap available to the Lua memory allocator.
+
+#### lua_gc
+
+`  int lua_gc (lua_State *L, int what, int data);`         [-0, +0, m]
+
+provides an option for 
+- `what`:
+    - `LUA_GCSETMEMLIMIT`  sets the available heap threshold (in bytes) at which aggressive sweeping starts.
+
+#### lua_getlfsconfig
+
+`  void lua_getlfsconfig (lua_State *L, int *conf);`         [-0, +0, -]
+
+if `conf` is not `NULL`, then this returns an int[5] summary of the LFS configuration.  The first 3 items are the mapped and flash address of the LFS region, and its allocated size.  If the LFS is loaded then the 4th is the current size used and the 5th (for Lua 5.3 only) the date-timestamp of loaded LFS image. 
+
+#### lua_getstate
+
+`  lua_State * lua_getstate();`         [-0, +0, -]
+
+returns the main thread `lua_State` record.  Used in CBs to initialise `L` for subsequent API call use.
+
+#### lua_pushrotable
+`  void lua_pushrotable (lua_State *L, ROTable *p);`         [-0, +1, -]
+
+Pushes a ROTable onto the stack.
+
+#### lua_pushstringsarray
+`  int lua_pushstringsarray (lua_State *L, int opt);`         [-0, +1, m]
+
+Pushes an array onto the stack containing all strings in the specified strings table. If `opt` = 0 then the RAM table is used, else if `opt` = 1 and LFS is loaded then the LFS  table is used, else `nil` is pushed onto the stack.
+
+Returns a status boolean 1 = table pushed. 
+
+#### lua_stripdebug
+`  int lua_stripdebug (lua_State *L, int level);`         [-1, +0, e]
+
+This function has two modes.  A value is popped off the stack.  
+-  If this value is `nil`, then the default strip level is set to this if `level` in the range 0 to 2, otherwise the current default level is returned.
```

Comment:
The current default level is always returned
```suggestion
-  If this value is `nil`, then the default strip level is set to this if `level` in the range 0 to 2. Returns the current default level.
```

Reply:
Agreed.  Also added the following to the next point `Returns an _approximate_ estimate of the heap freed (this can be a lot higher if some strings can be garbage collected).`  In practice because of heap management overheads etc. and GC of strings this estimate can be 2-3× too small.  IMO, better to get the heap size before and after or just avoid the allocation in the first place by setting the default level beforehand.